chess-AI/train.py

```diff
@@ -15,231 +15,209 @@
-                state_values = state_values.to(device=self.device)
-
-                # Compute policy loss and value loss using loss functions
-                pol_loss = ce_loss_fn(policy_batch, move_probs)
-                val_loss = mse_loss_fn(value_batch, state_values)
-                loss = pol_loss + val_loss
-
-                # Add to list for graphing purposes
-                policy_losses.append(pol_loss)
-                value_losses.append(val_loss)
-                losses.append(loss.item())
-
-                # Calculate gradients, update parameters, and reset gradients
-                loss.backward()
-                opt.step()
-                opt.zero_grad()
-
-            end = time.time()
-
-            # Calculate and store the average losses
-            policy_loss = sum(policy_losses) / len(policy_losses)
-            value_loss = sum(value_losses) / len(value_losses)
-            average_pol_loss.append(policy_loss.cpu().detach().numpy())
-            average_val_loss.append(value_loss.cpu().detach().numpy())
-
-
-
-        # Saves model to specified file, or a new file if not specified.
-        # TODO: Figure frequency of model saving, right now it is after every epoch
-        # TODO: Determine better names for the models when saving
-        if self.save_path is not None:
-            save_model(nnet, self.save_path)
-        else:
-            # Iterate through the number of models saved
-            for i in range(num_saved_models):
-                if not (os.path.isfile(f'models/models-{i + 1}.pt')):
-                    if overwrite_save and i != 0:
-                        save_model(nnet, f'models/models-{i}.pt')
-                        break
-                    save_model(nnet, f'models/models-{i + 1}.pt')
+def train_on_dataset(dataset, nnet, learning_rate, epochs, batch_size, device, save_path, num_saved_models, overwrite_save):
+    """Train with the specified dataset
+
+    Attributes:
+        dataset: the dataset to use for training
+        dashboard: the streamlit dashboard (this will eventually be removed)
+        nnet: the neural network
+        learning_rate: the learning rate for the optimizer
+        epochs: number of epochs
+        batch_size: the batch size for SGD
+        device: the device being used to train (either CPU or GPU)
+        save_path: path for model checkpointing
+        num_saved_models: number of models to store
+    """
+
+    # Stores the average losses which are used for graphing
+    average_pol_loss = []
+    average_val_loss = []
+
+    # Define loss functions (cross entropy and mean squared error)
+    ce_loss_fn = torch.nn.CrossEntropyLoss()
+    mse_loss_fn = torch.nn.MSELoss()
+
+    # Create optimizer for updating parameters during training
+    # TODO: Consider using other optimizers, such as Adam
+    opt = torch.optim.SGD(nnet.parameters(), lr=learning_rate, weight_decay=0.001, momentum=0.9)
```

Comment:
Again, I know this is how it was before the PR, but should we pass `weight_decay` and `momentum` as parameters?